### PR TITLE
Update individual's path to .d4 files for an existing case

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Use the cli to update the path to .d4 files for individuals of an existing case (moving towards chanjo2 - #4000)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Locally, run the following command: `scout --demo update individual -c internal_id -n NA12882 d4_file <path to file>`

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
image


Before:

image


Update via cli: 🆗 

image


image

File: scout/commands/update/individual.py [hunk 1]
```diff
@@ -10,24 +10,25 @@
     "bam_file": "path",
     "bionano_access.sample": "str",
     "bionano_access.project": "str",
+    "d4_file": "path",
```
Comment (northwestwitch) [inline scout/commands/update/individual.py:13]: New type!

File: tests/commands/update/test_update_individual_cmd.py [hunk 2]
```diff
@@ -96,8 +97,11 @@ def test_update_key_non_existent_path(mock_app, real_populated_database, case_ob
     assert "The provided path was not found on the server, update key anyway?" in result.output
 
 
-def test_update_alignment_path(mock_app, real_populated_database, bam_path):
-    """Tests the CLI that updates a individual with an alignment path"""
+@pytest.mark.parametrize("update_key", UPDATE_DICT)
```
Comment (dnil) [inline tests/commands/update/test_update_individual_cmd.py:100]: 🥇

File: tests/commands/update/test_update_individual_cmd.py [hunk 3]
```diff
@@ -107,14 +111,25 @@ def test_update_alignment_path(mock_app, real_populated_database, bam_path):
...
     )
 
     # THEN the command should run with no errors
     assert result.exit_code == 0
-    # THEN assert that the new alignment path was added
+
+    # THEN the individual should be updated
     fetched_case = real_populated_database.case_collection.find_one()
-    assert fetched_case["individuals"][0]["bam_file"] == str(bam_path)
+    if "." in update_key:
```
Comment (dnil) [inline tests/commands/update/test_update_individual_cmd.py:128]: 👍

**Review by dnil — APPROVED** [review]
👍

**Pull request merged**